src/test/libcephfs/test.cc

```diff
@@ -3793,3 +3793,39 @@ TEST(LibCephFS, SnapdirAttrsOnSnapRename) {
   ASSERT_EQ(0, ceph_unmount(cmount));
   ceph_shutdown(cmount);
 }
+
+TEST(LibCephFS, PathWalkAndLookup) {
+  struct ceph_mount_info *cmount;
+  ASSERT_EQ(ceph_create(&cmount, NULL), 0);
+  ASSERT_EQ(ceph_conf_read_file(cmount, NULL), 0);
+  ASSERT_EQ(0, ceph_conf_parse_env(cmount, NULL));
+  ASSERT_EQ(ceph_mount(cmount, NULL), 0);
+
+  int ret;
+  char dir_name[128];
+  char dir_path[256];
+
+  pid_t mypid = getpid();
+  sprintf(dir_name, "dir_%d", mypid);
+  sprintf(dir_path, "/%s", dir_name);
+
+  ASSERT_EQ(ceph_mkdir(cmount, dir_path, 0777), 0);
```

Comment:
do we need to create this dir? we are making use of dir_name instead of dir_path in the last lookup (because we want ENOENT), i guess we can just omit this.

Reply:
ACK.

Reply:
err, actually I wanted to create the file and check if we really get back an ENOENT.